File: droidboy/src/main/java/com/appboy/sample/FullyCustomNotificationFactory.java
package com.appboy.sample;

import android.app.Notification;
import android.os.Bundle;

import androidx.core.app.NotificationCompat;

import com.appboy.models.push.BrazeNotificationPayload;
import com.braze.IBrazeNotificationFactory;
import com.braze.push.BrazeNotificationUtils;

public class FullyCustomNotificationFactory implements IBrazeNotificationFactory {

  @Override
  public Notification createNotification(BrazeNotificationPayload payload) {
    String notificationChannelId = BrazeNotificationUtils.getOrCreateNotificationChannelId(payload);
    NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(payload.getContext(), notificationChannelId);
    notificationBuilder.setContentTitle(payload.getTitleText());
    notificationBuilder.setSmallIcon(R.drawable.com_braze_push_small_notification_icon);
    BrazeNotificationUtils.setAccentColorIfPresentAndSupported(notificationBuilder, payload);
    String contentString = parseContentsFromExtras(payload.getBrazeExtras());
    notificationBuilder.setContentText(contentString);
    return notificationBuilder.build();
  }

  private String parseContentsFromExtras(Bundle appboyExtras) {
    String buildString = "Your order: ";
    buildString += appboyExtras.getString(PushTesterFragment.EXAMPLE_APPBOY_EXTRA_KEY_1);
    buildString += ", ";
    buildString += appboyExtras.getString(PushTesterFragment.EXAMPLE_APPBOY_EXTRA_KEY_2);
    buildString += ", and ";
    buildString += appboyExtras.getString(PushTesterFragment.EXAMPLE_APPBOY_EXTRA_KEY_3);
    buildString += ".";
    return buildString;
  }
}

File: droidboy/src/main/java/com/appboy/sample/FullyCustomNotificationFactory.kt
package com.appboy.sample

import android.app.Notification
import androidx.core.app.NotificationCompat
import com.appboy.models.push.BrazeNotificationPayload
import com.braze.IBrazeNotificationFactory
import com.braze.push.BrazeNotificationUtils.getOrCreateNotificationChannelId
import com.braze.push.BrazeNotificationUtils.setAccentColorIfPresentAndSupported

class FullyCustomNotificationFactory : IBrazeNotificationFactory {
    override fun createNotification(payload: BrazeNotificationPayload): Notification? {
        val context = payload.context ?: return null

        val notificationChannelId = getOrCreateNotificationChannelId(payload)
        val notificationBuilder = NotificationCompat.Builder(context, notificationChannelId)
        notificationBuilder.setContentTitle(payload.titleText)
        notificationBuilder.setSmallIcon(R.drawable.com_braze_push_small_notification_icon)
        setAccentColorIfPresentAndSupported(notificationBuilder, payload)
        val contentString = parseContentsFromExtras(payload.extras)
        notificationBuilder.setContentText(contentString)
        return notificationBuilder.build()
    }

    private fun parseContentsFromExtras(extras: Map<String, String>): String = """
        Your order: ${extras[PushTesterFragment.EXAMPLE_APPBOY_EXTRA_KEY_1]}, 
        ${extras[PushTesterFragment.EXAMPLE_APPBOY_EXTRA_KEY_2]}, 
        ${extras[PushTesterFragment.EXAMPLE_APPBOY_EXTRA_KEY_3]}.
    """.trimIndent()
}
